### PR TITLE
Use enums as sorting arguments

### LINKED ISSUE #3300: Use enums as sorting arguments
In queries, the supported sorting options should be represented as enums. Now we use strings, so API consumers are not able to determine what options are available.

Currently, this issue applies only to one query:
- `products(sortBy: String)`

Available sorting fields (based on Dashboard 2.0 and storefront) are now: `price`, `name`.

Schema that would solve the issue (inspired by Github API v4):
```
enum ProductOrderField {
  NAME
  PRICE
}

enum OrderDirection {
  ASC
  DESC
}

type ProductOrder {
  field: ProductOrderField!
  direction: OrderDirection!
}

type Query {
  products(sortBy: ProductOrder)
}
```

### PR DESCRIPTION
**Pull request by fowczarek**
I want to merge this change because resolve #3300

### Pull Request Checklist

1. [x] Privileged views and APIs are guarded by proper permission checks.
1. [x] All visible strings are translated with proper context.
1. [x] All data-formatting is locale-aware (dates, numbers, and so on).
1. [x] Database queries are optimized and the number of queries is constant.
1. [x] The changes are tested.
1. [x] The code is documented (docstrings, project documentation).
1. [x] GraphQL schema and type definitions are up to date.

### CODE REVIEW

File: saleor/graphql/product/types.py [hunk 1]
```diff
@@ -129,6 +139,15 @@ class Meta:
         description = 'Represents a custom attribute.'
 
 
+class ProductOrder(graphene.InputObjectType):
+    field = graphene.Argument(
+        ProductOrderField, required=True,
+        description='Allows selecting field to sort.')
```
Comment (maarcingebala) [inline saleor/graphql/product/types.py]: "Sort products by the selected field."

File: saleor/graphql/product/filters.py [hunk 2]
```diff
@@ -55,6 +55,8 @@ def filter_products_by_collections(qs, collections):
 
 
 def sort_qs(qs, sort_by):
```
Comment (NyanKiyoshi) [inline saleor/graphql/product/filters.py]: Could we add a type hint in the signature? `sort_by` sounds like a string to me.

File: saleor/graphql/product/types.py [hunk 3]
```diff
@@ -129,6 +139,15 @@ class Meta:
...
+    direction = graphene.Argument(
+        OrderDirection, required=True,
+        description='Determines how the field is being sorted.')
```
Comment (maarcingebala) [inline saleor/graphql/product/types.py]: "Specifies the direction in which to sort products"

File: saleor/graphql/product/types.py [hunk 4]
```diff
@@ -42,6 +42,16 @@ class StockAvailability(graphene.Enum):
     OUT_OF_STOCK = 'OUT_OF_STOCK'
 
 
+class ProductOrderField(graphene.Enum):
```
Comment (maarcingebala) [inline saleor/graphql/product/types.py]: Graphene allows adding descriptions for enum values as well - https://docs.graphene-python.org/en/latest/types/enums/#value-descriptions

Here I would add:
- name - "Sort products by name"
- price = "Sort products by price"

File: saleor/graphql/product/types.py [hunk 5]
```diff
@@ -42,6 +42,16 @@ class StockAvailability(graphene.Enum):
...
+    NAME = 'name'
+    PRICE = 'price'
+
+
+class OrderDirection(graphene.Enum):
```
Comment (maarcingebala) [inline saleor/graphql/product/types.py]: Let's add descriptions for these values as well:
- ASC - "Specifies an ascending sort order."
- DESC - "Specifies a descending sort order."

**Pull request merged**